File: sale-process/sales-app/src/app/interfaces/detalle-venta.ts
import { Producto } from './producto';
import { Venta } from './venta';

export interface DetalleVenta {
  id: number;
  venta: Venta | null; 
  producto: Producto;
  cantidad: number;
  precioUnitario: number;
  desconto: number;
  subtotal: number;
  igv: number;
  total: number;
  estado: boolean;
}

File: sale-process/sales/src/main/java/sales/system/sales/model/DetalleVenta.java
package sales.system.sales.model;

import com.fasterxml.jackson.annotation.JsonBackReference;
import lombok.*;
import jakarta.persistence.*;

@Data
@Entity
public class DetalleVenta {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "venta_id")
    @JsonBackReference  // Evita la recursión al no serializar esta propiedad
    private Venta venta;

    @ManyToOne
    @JoinColumn(name = "producto_id")
    private Producto producto;

    private Integer cantidad;
    private Double precioUnitario;
    private Double subtotal;
    private Double desconto;
    private Double igv;
    private Double total;

    private Boolean estado = true;
}
